File: Problem-036/solution.py
def double_base_palindromes(n):
    total = 0

    for i in range(1, n):
        if palindrome(i) and palindrome(to_binary(i)):
            total += i

    return total


def to_binary(num):
    return bin(num)[2:]


def palindrome(value):
    string = str(value)
    return string == string[::-1]


print(double_base_palindromes(1_000_000))

File: Problem-036/solution.js
function doubleBasePalindromes(n) {
    let total = 0;

    for (let i = 1; i < n; i++) {
        if (pali(i) && pali(toBinary(i))) {
            total += i;
        }
    }

    return total;
}


function toBinary(m) {
    return m.toString(2);
}


function pali(p) {
    let str = p.toString();

    return str === str.split('').reverse().join('');
}
